### PR TITLE
feat(upgrade): upgrading a study also upgrades all of its children

### PR DESCRIPTION
**Pull request by MartinBelthle**
Linked to this issue : https://github.com/AntaresSimulatorTeam/AntaREST/issues/1212 but i cannot link them i do not know why

### CODE REVIEW

**Comment by ghost** [comment]
### Features

*  **upgrade:** upgrading a study also upgrades all of its children ([605897d](https://github.com/AntaresSimulatorTeam/AntaREST/commit/605897d6dfaa7d5c7596d279b4c52ac1347ff560))


### Bug Fixes

*  **tests:** rollback test_variant_model.py ([96ab67d](https://github.com/AntaresSimulatorTeam/AntaREST/commit/96ab67d6167d08ed2ce6babd34812afd0b4a79e7))
*  **upgrade:** clear snapshot when upgrading a variant as it used to be done before ([15ddaa4](https://github.com/AntaresSimulatorTeam/AntaREST/commit/15ddaa42fa319028b500b5d43063735efffb3525))
*  **imports:** remove unused import ([0fb1469](https://github.com/AntaresSimulatorTeam/AntaREST/commit/0fb1469dc61e5f7d77da3ece6a23015c530374e0))


### Contributors

<a href="https://github.com/MartinBelthle">MartinBelthle</a>

<details>
<summary>Commit-Lint commands</summary>
<br />

You can trigger Commit-Lint actions by commenting on this PR:
- `@Commit-Lint merge patch` will merge dependabot PR on "patch" versions (X.X.Y - Y change)
- `@Commit-Lint merge minor` will merge dependabot PR on "minor" versions (X.Y.Y - Y change)
- `@Commit-Lint merge major` will merge dependabot PR on "major" versions (Y.Y.Y - Y change)
- `@Commit-Lint merge disable` will desactivate merge dependabot PR
- `@Commit-Lint review` will approve dependabot PR
- `@Commit-Lint stop review` will stop approve dependabot PR



</details>

File: antarest/study/service.py [hunk 1]
```diff
@@ -234,11 +234,26 @@ def _upgrade_study(self) -> None:
...
                 else:
-                    study_path = Path(study_to_upgrade.path)
-                    upgrade_study(study_path, target_version)
-                remove_from_cache(self.cache_service, study_to_upgrade.id)
+                    upgrade_study(Path(study_to_upgrade.path), target_version)
+                    remove_from_cache(self.cache_service, study_to_upgrade.id)
                 study_to_upgrade.version = target_version
                 self.repository.save(study_to_upgrade)
+
+                variants = self.storage_service.variant_study_service.repository.get_children(
```
Comment (laurent-laporte-pro) [inline antarest/study/service.py:242]: To simplify the algorithm, you should consider a recursive approach. Here, we are changing the study version, but it's also necessary to upgrade the commands in the study history.

**Review by laurent-laporte-pro — CHANGES_REQUESTED** [review]
Upgrading a study involves not only changing the version of a study but also modifying its properties, and in some cases, the matrices of the study. For a RAW study, it's sufficient to modify the files on disk, but for study variants, it's more complex: we need to consider upgrading some commands in the study history. It's possible that commands may operate on older properties or matrices, and they may also need modification to accommodate the format change.

**Martin**:
Indeed, you are right. I only considered commands  `create` and `remove` for areas and links that are versioned so I thought it was okay. But modifying some matrix in a link, raises a command `replace_matrix` which does not work when we upgrade a study from 8.1 to 8.2. Moreover, as the matrix is splitted in 3 different matrices, the command should become 3 distinct ones :confused: . So i think handling such a case is really complicated and takes a lot of time to develop. Plus it is not really important as users use almost only v8.2 and newer versions.

**Comment by MartinBelthle** [comment]
Le problème est plus compliqué que prévu et cette PR ne le résolvera pas. Je la close car elle ne fix pas le souci

**Pull request closed without merging**